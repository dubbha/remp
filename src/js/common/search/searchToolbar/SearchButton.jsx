import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

export default class SearchButton extends Component {
  handleClick = () => {
    const { onClickWithText, text } = this.props;

    if (onClickWithText) {
      onClickWithText(text);
    }
  }

  render() {
    const {
      text,
      type,
      size,
      active,
    } = this.props;

    return (
      <button
        type={type}
        className={classNames(
          'searchButton',
          {
            [`searchButton_${size}`]: true,
            searchButton_active: active,
          },
        )}
        onClick={this.handleClick}
      >
        {`${text}`}
      </button>
    );
  }
}

SearchButton.defaultProps = {
  type: 'button',
  size: 'small',
  active: false,
  onClickWithText: null,
};

SearchButton.propTypes = {
  text: PropTypes.string.isRequired,
  type: PropTypes.string,
  size: PropTypes.string,
  active: PropTypes.bool,
  onClickWithText: PropTypes.func,
};
