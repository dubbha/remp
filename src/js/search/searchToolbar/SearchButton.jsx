import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const SearchButton = ({
  text,
  type,
  size,
  active,
}) => (
  <button
    type={type}
    className={classNames(
      'searchButton',
      {
        [`searchButton_${size}`]: true,
        searchButton_active: active,
      },
    )}
  >
    {`${text}`}
  </button>
);

SearchButton.defaultProps = {
  type: 'button',
  size: 'small',
  active: false,
};

SearchButton.propTypes = {
  text: PropTypes.string.isRequired,
  type: PropTypes.string,
  size: PropTypes.string,
  active: PropTypes.bool,
};

export default SearchButton;
