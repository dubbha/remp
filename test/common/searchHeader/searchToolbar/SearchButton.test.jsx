import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SearchButton from 'common/searchHeader/searchToolbar/SearchButton';

describe('SearchButton', () => {
  let props;

  beforeEach(() => {
    props = {
      text: 'TEXT',
    };
  });

  it('should render successfully with defaults', () => {
    const wrapper = shallow(<SearchButton {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render successfully with passed values', () => {
    props = {
      ...props,
      type: 'submit',
      size: 'large',
      active: true,
      onClickWithText: jest.fn(),
    };
    const wrapper = shallow(<SearchButton {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should handle click if on click function is passed', () => {
    props.onClickWithText = jest.fn();
    const wrapper = shallow(<SearchButton {...props} />);

    wrapper.find('button').simulate('click');

    expect(props.onClickWithText).toBeCalled();
  });

  it('should silently ignore the click otherwise', () => {
    const wrapper = shallow(<SearchButton {...props} />);

    // Wrap in a function, otherwise the error will not be caught and the assertion will fail
    const wrapFn = () => wrapper.find('button').simulate('click');

    expect(wrapFn).not.toThrow();
  });
});
