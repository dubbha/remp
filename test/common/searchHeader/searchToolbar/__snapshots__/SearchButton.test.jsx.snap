// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchButton should render successfully with defaults 1`] = `
<button
  className="searchButton searchButton_small"
  onClick={[Function]}
  type="button"
>
  TEXT
</button>
`;

exports[`SearchButton should render successfully with passed values 1`] = `
<button
  className="searchButton searchButton_large searchButton_active"
  onClick={[Function]}
  type="submit"
>
  TEXT
</button>
`;
